Examples/Algorithms/Geant4/src/TelescopeDetectorConstruction.cpp

```diff
@@ -0,0 +1,122 @@
+#include "ActsExamples/TelescopeDetector/TelescopeDetectorConstruction.hpp"
+
+#include "Acts/Definitions/Units.hpp"
+#include "Acts/Utilities/BinningType.hpp"
+#include "ActsExamples/TelescopeDetector/BuildTelescopeDetector.hpp"
+
+#include "G4Box.hh"
+#include "G4Cons.hh"
+#include "G4LogicalVolume.hh"
+#include "G4NistManager.hh"
+#include "G4Orb.hh"
+#include "G4PVPlacement.hh"
+#include "G4RunManager.hh"
+#include "G4Sphere.hh"
+#include "G4SystemOfUnits.hh"
+#include "G4Trd.hh"
+#include "globals.hh"
+
+ActsExamples::Telescope::TelescopeG4DetectorConstruction::
+    TelescopeG4DetectorConstruction(TelescopeDetector::Config cfg)
+    : m_cfg(cfg) {
+  throw_assert(cfg.surfaceType ==
+                   static_cast<int>(Telescope::TelescopeSurfaceType::Plane),
+               "only plan is supported right now");
+}
+
+G4VPhysicalVolume*
+ActsExamples::Telescope::TelescopeG4DetectorConstruction::Construct() {
+  G4double length = m_cfg.positions.back() - m_cfg.positions.front();
+
+  // Get nist material manager
+  G4NistManager* nist = G4NistManager::Instance();
+
+  // Envelope parameters
+  //
+  G4double env_sizeX = m_cfg.bounds[0] + 5. * mm;
+  G4double env_sizeY = m_cfg.bounds[1] + 5. * mm;
+  G4double env_sizeZ = length + 10. * mm;
+
+  // Option to switch on/off checking of volumes overlaps
+  //
+  G4bool checkOverlaps = true;
+
+  // Materials
+  G4Material* galactic = nist->FindOrBuildMaterial("G4_Galactic");
+  G4Material* silicon = nist->FindOrBuildMaterial("G4_Si");
+
+  //
+  // World
+  //
+  G4double worldSizeX = 2 * m_cfg.bounds[0];
+  G4double worldSizeY = 2 * m_cfg.bounds[1];
+  G4double worldSizeZ = length;
+
+  // Construct the rotation
+  // This assumes the binValue is binX, binY or binZ. No reset is necessary in
+  // case of binZ
+  G4RotationMatrix* rotation = nullptr;
+  if (m_cfg.binValue == Acts::BinningValue::binX) {
```

Comment:
I'm actually not entirely happy with comparing an `int` with an `Acts::BinningValue` (which is an enum)...
But maybe it would make more sense to change the `Acts::BinningValue` to a scoped enum?

But for the start it could also be enough to check if the value is 0, 1 or 2 and trow otherwise?

Reply:
same :smile: I think the reason is that this config value comes from python? I will change it to an enum class and use static cast

Reply:
Yeah probably. The cleaner way would be to bind the `enum` properly to python, so that also in python we could pass something like `acts.BinningValue.binX` to the config...
But not sure if that must happen in that PR